impeller/entity/shaders/linear_gradient_fill.frag

```diff
@@ -30,5 +31,5 @@ void main() {
     gradient_info.texture_sampler_y_coord_scale,
     gradient_info.tile_mode,
     gradient_info.tile_mode);
-  frag_color = vec4(frag_color.xyz * frag_color.a, frag_color.a);
+  frag_color = vec4(frag_color.xyz * frag_color.a, frag_color.a) * gradient_info.alpha;
```

Comment:
I left this comment on the other PR, but this still makes me wonder if `IPSampleWithTileMode` shouldn't really just be `IPSampleWithTileModeAndAlpha` so we don't forget to do this somewhere - e.g. if and when we implement two point conical gradients.

Reply:
One idea for making this a bit more semantic is to add an `IPPremultiply` utility to the shader library and wrap the sampling function in it. This way new contributors who don't know why we're doing the funky alpha multiplies can get a bit more context when first encountering it.